nifi-framework-bundle/nifi-framework/nifi-framework-components/src/main/java/org/apache/nifi/groups/StandardProcessGroup.java

```diff
@@ -4040,14 +4045,20 @@ public void verifyCanSaveToFlowRegistry(final String registryId, final String bu
 
                 // Flow ID matches. We want to publish the Process Group as the next version of the Flow, so we must
                 // ensure that all other parameters match as well.
+
+                if (branch != null && !Objects.equals(branch, vci.getBranch())) {
```

Comment:
This two statement could be merged into one with the following way:
- `FlowLocation` might be able to proviede a `BucketLocation` instance (like a partial path)
- We could compare `BucketLocation` instnaces
I think that would simplify this, might communicate the intent in a more accurate manner and as we have the same error message we would not even loose details